common/src/reserved.rs

```diff
@@ -59,18 +62,72 @@ impl ReservedState {
                     .or_insert(member.consensus_voting_power);
             }
         }
-        Ok(governance_set
-            .iter()
-            .map(|(name, voting_power)| (self.query_public_key(name).unwrap(), *voting_power))
-            .collect())
+        let mut result = Vec::new();
+        for (name, voting_power) in governance_set {
+            let public_key = self.query_public_key(&name).ok_or_else(|| {
+                format!("The public key of {name} is not found in the genesis info.")
+            })?;
+            result.push((public_key, voting_power));
+        }
+        Ok(result)
     }
 
-    pub fn apply_delegate(&mut self, _tx: &TxDelegate) -> Result<Self, String> {
-        unimplemented!()
+    pub fn apply_delegate(&mut self, tx: &TxDelegate) -> Result<Self, String> {
+        let data = DelegationTransactionData {
+            delegator: tx.delegator.clone(),
+            delegatee: tx.delegatee.clone(),
+            governance: tx.governance,
+            timestamp: tx.timestamp as BlockHeight,
```

Comment:
`tx.timestamp` is not `BlockHeight`. It's literally timestamp of the transaction.

Reply:
I made the changes as you told me in Discord - the last field of two structs is `block_height` which is `BlockHeight` type.

```rust
#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
pub struct DelegationTransactionData {
    pub delegator: PublicKey,
    pub delegatee: PublicKey,
    pub governance: bool,
    pub block_height: BlockHeight,
}

#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
pub struct UndelegationTransactionData {
    pub delegator: PublicKey,
    pub block_height: BlockHeight,
}
```